webapp/channels/src/components/integrations/installed_incoming_webhooks/installed_incoming_webhooks.tsx

```diff
@@ -44,26 +48,43 @@ export default class InstalledIncomingWebhooks extends React.PureComponent<Props
         super(props);
 
         this.state = {
+            page: 0,
             loading: true,
         };
     }
 
     componentDidMount() {
-        if (this.props.enableIncomingWebhooks) {
-            this.props.actions.loadIncomingHooksAndProfilesForTeam(
-                this.props.team.id,
-                Constants.Integrations.START_PAGE_NUM,
-                Constants.Integrations.PAGE_SIZE,
-            ).then(
-                () => this.setState({loading: false}),
-            );
-        }
+        this.loadPage(0);
     }
 
     deleteIncomingWebhook = (incomingWebhook: IncomingWebhook) => {
         this.props.actions.removeIncomingHook(incomingWebhook.id);
     };
 
+    loadPage = async (pageToLoad: number) => {
+        if (this.props.enableIncomingWebhooks) {
+            this.setState({loading: true},
+                async () => {
+                    await this.props.actions.loadIncomingHooksAndProfilesForTeam(
```

Comment:
please wrap this in a try-catch-finally block to handle cases where api request fail.

Reply:
Really interesting point.

I took a closer look at this. When an API request fails, the error is caught in `IntegrationActions.getIncomingHooks` which is called by `loadIncomingHooksAndProfilesForTeam`. The `loadIncomingHooksAndProfilesForTeam` function also calls `loadProfilesForIncomingHooks`, which could also cause the API request to fail. But in this case all errors will be caught in the called `getProfilesByIds` method. I also tried to simulate an API request fail. This does indeed cause the error to be caught in `IntegrationActions.getIncomingHooks`.

So I don't see what error (that is not yet caught by any of the called functions) could be caught in the suggested try-catch-finally block. Do you have any other errors in mind that I missed?

Reply:
just for good practice it is a good idea to wrap it, think that now the async functions being called have their set error handling, but any new code added in future might not, also there can always be non-expected errors that are not related to the API requests(network issues, timeouts, or runtime errors are the ones that I can think of now). And the finally also will make sure that this.setState({loading: false}) keeps consistent accordingly with the state of the request.